doc/tutorials/tutorial_understanding_building_benchmark.rst

```diff
@@ -0,0 +1,370 @@
+            'scale_step': [1, 1.99]
+        }
+
+Updating the template to implement a ridge regression benchmark
+---------------------------------------------------------------
+
+We are now equipped with enough knowledge to update the template benchmark to a ridge regression benchmark.
+Formally, we are starting from OLS and GD implemented for the OLS problem.
+Therefore we need to implement the following modifications:
+
+    - we should add the regularization term \\( +\\lambda \\|\\beta \\|^2 \\) to the loss in ``objective.py``, and values for the regularization parameter.
+    - we should modify the computed gradient, knowing that \\( \\nabla f(\\beta) = \\nabla g(\\beta) + 2\\lambda\\beta \\).
+
+We will not modify anything in the dataset since the inputs \\(X,y \\) of the regression and ridge regression are essentially the same.
+
+Let's start with the ``objective.py`` file.
+The regularization parameter values are part of the formal definition of the problem, so we can define them as attributes of the ``Objective`` class.
+The ``whiten_y`` parameter is already implemented, so we can simply add a ``reg`` parameter in the ``parameters`` dictionary.
+
+.. code-block:: python
+
+    class Objective(BaseObjective):
+        name = "Ordinary Least Squares"
+        parameters = {
+            'whiten_y': [False, True],
+            'reg': [1e1, 1e2]
+        }
+
+This piece of code says that \\( \\lambda\\) should take two values \\( 10\\) or \\( 100\\) across the benchmark.
+
+Then we update the ``compute`` method as follows:
+
+.. code-block:: python
+
+        def compute(self, beta):
+            diff = self.y - self.X.dot(beta)
+            l2reg = self.reg*np.linalg.norm(beta)**2
+            return dict(
+                value=.5 * diff.dot(diff) + l2reg,
+                ols=.5 * diff.dot(diff),
+                penalty=l2reg
+            )
+
+We have done several modifications here:
+
+    - The ``l2reg`` variable computes the regularization term. It is added to the OLS term in the ``value`` field of the output dictionary. This ``value`` field is the main loss of the benchmark, used by all algorithms to track convergence.
+    - Additional metrics are computes, namely ``ols`` and ``penalty``. Benchmark will compute these metrics alongside the loss function, and we will be able to look at them in the resulting plots.
+
+One additional modification handles the fact that the solvers will require the knowledge of \\(\\lambda\\).
+The way to communicate from objectives to solvers, according to the figure above, is by using the ``get_objective`` method.
+It can be modified as follows
+
+.. code-block:: python
+
+        def get_objective(self):
+            return dict(
+                X=self.X,
+                y=self.y,
+                reg=self.reg
+            )
+
+That's it for the ``objective.py`` file!
+We can now modify the solver.
+Instead of modifying directly ``python-gd.py``, let's create a new solver, ``python-gd-ridge.py``.
```

Comment:
As said above, I'd modify this solver instead, keeping in mind that it's for beginners.

Reply:
Done, only one solver and we use the same file